binding/Binding/SKFont.cs

```diff
@@ -0,0 +1,718 @@
+
+		protected override void DisposeNative () =>
+			SkiaApi.sk_font_delete (Handle);
+
+		public bool ForceAutoHinting {
+			get => SkiaApi.sk_font_is_force_auto_hinting (Handle);
+			set => SkiaApi.sk_font_set_force_auto_hinting (Handle, value);
+		}
+
+		public bool EmbeddedBitmaps {
+			get => SkiaApi.sk_font_is_embedded_bitmaps (Handle);
+			set => SkiaApi.sk_font_set_embedded_bitmaps (Handle, value);
+		}
+
+		public bool Subpixel {
+			get => SkiaApi.sk_font_is_subpixel (Handle);
+			set => SkiaApi.sk_font_set_subpixel (Handle, value);
+		}
+
+		public bool LinearMetrics {
+			get => SkiaApi.sk_font_is_linear_metrics (Handle);
+			set => SkiaApi.sk_font_set_linear_metrics (Handle, value);
+		}
+
+		public SKFontHinting HintingLevel {
+			get => SkiaApi.sk_font_get_hinting (Handle);
+			set => SkiaApi.sk_font_set_hinting (Handle, value);
+		}
+
+		public bool Embolden {
+			get => SkiaApi.sk_font_is_embolden (Handle);
+			set => SkiaApi.sk_font_set_embolden (Handle, value);
+		}
+
+		public bool BaselineSnap {
+			get => SkiaApi.sk_font_is_baseline_snap (Handle);
+			set => SkiaApi.sk_font_set_baseline_snap (Handle, value);
+		}
+
+		public SKFontEdging Edging {
+			get => SkiaApi.sk_font_get_edging (Handle);
+			set => SkiaApi.sk_font_set_edging (Handle, value);
+		}
+
+		public SKFontHinting Hinting {
+			get => SkiaApi.sk_font_get_hinting (Handle);
+			set => SkiaApi.sk_font_set_hinting (Handle, value);
+		}
+
+		public SKTypeface Typeface {
+			get => GetObject<SKTypeface> (SkiaApi.sk_font_get_typeface (Handle));
+			set => SkiaApi.sk_font_set_typeface (Handle, value == null ? IntPtr.Zero : value.Handle);
+		}
+
+		public float Size {
+			get => SkiaApi.sk_font_get_size (Handle);
+			set => SkiaApi.sk_font_set_size (Handle, value);
+		}
+
+		public float ScaleX {
+			get => SkiaApi.sk_font_get_scale_x (Handle);
+			set => SkiaApi.sk_font_set_scale_x (Handle, value);
+		}
+
+		public float SkewX {
+			get => SkiaApi.sk_font_get_skew_x (Handle);
+			set => SkiaApi.sk_font_set_skew_x (Handle, value);
+		}
+
+		// FontMetrics
+
+		public float Spacing =>
+			SkiaApi.sk_font_get_metrics (Handle, null);
+
+		public SKFontMetrics Metrics {
+			get {
+				GetFontMetrics (out var metrics);
+				return metrics;
+			}
+		}
+
+		public float GetFontMetrics (out SKFontMetrics metrics)
+		{
+			fixed (SKFontMetrics* m = &metrics) {
+				return SkiaApi.sk_font_get_metrics (Handle, m);
+			}
+		}
+
+		// GetGlyph
+
+		public ushort GetGlyph (int codepoint) =>
+			SkiaApi.sk_font_unichar_to_glyph (Handle, codepoint);
+
+		// GetGlyphs
+
+		public ushort[] GetGlyphs (ReadOnlySpan<int> codepoints)
+		{
+			var glyphs = new ushort[codepoints.Length];
+			GetGlyphs (codepoints, glyphs);
+			return glyphs;
+		}
+
+		public void GetGlyphs (ReadOnlySpan<int> codepoints, Span<ushort> glyphs)
+		{
+			fixed (int* up = codepoints)
+			fixed (ushort* gp = glyphs) {
+				SkiaApi.sk_font_unichars_to_glyphs (Handle, up, codepoints.Length, gp);
+			}
```

Comment:
This is option B, no allocations internally. All memory must be provided and it is then used as is.

Potential usage with `ArrayPool`:
```csharp
// magic some codepoints up
int[] codepoints = GetCodepoints ();

// get from the pool
var glyphs = ArrayPool<ushort>.Shared.Rent (codepoints.Length);

var font = new SKFont ();
font.GetGlyphs (codepoints, glyphs);

ProcessGlyphs (glyphs);

// return to pool
ArrayPool<ushort>.Shared.Return (glyphs);
```